sunpy/visualization/animator/wcs.py

```diff
@@ -299,10 +308,28 @@ def plot_start_image_2d(self, ax):
 
         return im
 
+    def get_2d_plot_limts(self):
+        """
+        Get vmin, vmax of a data slice when "vmax" or "vmin" is set to "auto".
+        """
+        vmin, vmax = None, None
+        if self.imshow_kwargs.get('vmin') == 'auto':
+            percent_limits = ((1, 99)*u.percent).to('%').value
+            vmin, _ = AsymmetricPercentileInterval(*percent_limits).get_limits(self.data_transposed)
```

Comment:
Why not work this out once and then set the return value based on the keyword argument?

Reply:
Sorry @nabobalis, not quite sure what you meant by this.

Reply:
This blocks of code are repeated with only the value you return changed. Why not combine them?

Reply:
To be clear, the issue is that if both `vmin` and `vmax` are set to "auto", you compute the limits twice unnecessarily with the repeated call.  It'd be cleaner to do something like:
```python
        if self.imshow_kwargs.get('vmin') == 'auto' or self.imshow_kwargs.get('vmax') == 'auto':
            percent_limits = ((1, 99)*u.percent).to('%').value
            vmin_temp, vmax_temp = AsymmetricPercentileInterval(*percent_limits).get_limits(self.data_transposed)
        vmin = vmin_temp if self.imshow_kwargs.get('vmin') == 'auto' else None
        vmax = vmax_temp if self.imshow_kwargs.get('vmax') == 'auto' else None
        return vmin, vmax
```